internal/core/src/storage/ThreadPool.h

```diff
@@ -60,6 +63,12 @@ class ThreadPool {
     void
     ShutDown();
 
+    size_t
+    GetThreadNum() {
+        std::lock_guard<std::mutex> lock(mutex_);
+        return current_threads_size_;
```

Comment:
how about https://en.cppreference.com/w/cpp/thread/shared_mutex

Reply:
you can  modify this in next pr. Since this GetThreadNum is not called by uppper layer.